base/linalg/dense.jl

```diff
@@ -310,15 +310,23 @@ kron(a::AbstractMatrix, b::AbstractVector)=kron(a,reshape(b,length(b),1))
 kron(a::AbstractVector, b::AbstractMatrix)=kron(reshape(a,length(a),1),b)
 
 # Matrix power
+function ^{T}(A::AbstractMatrix{T}, p::Integer)
+    # non-inversible matrix to negative power is undefined.
+    if (p < 0) && (det(A) == 0)
```

Comment:
This check does not seem like a good idea.  First, `inv(A)` will already throw an error if `A` is singular, and just as quickly as `det` will.  Second, computing `det(A)` means that you compute the LU factorization of `A` twice.   Third, you could get false positives because of underflow (`det([1e-300 0; 0 1e-300]) == 0` even though the matrix is not singular.

(Basically, you should forget everything you learned in high school about the determinant as a computational tool.)